src/Utils.jl

```diff
@@ -116,9 +116,13 @@ function _add_idmap_to_call(def::Expr, id_map::Union{Symbol,Expr})
     return Expr(:call, def.args[1], def.args[2:end]..., id_map)
 end
 
-@inline function fill_similar(value, array, args...)
+@inline function fill_similar(value::T, array, args...) where {T}
     out_array = similar(array, args...)
-    out_array .= value
+    if T<:Number
+        out_array .= value
+    else
+        fill!(out_array, value)
```

Comment:
Could we use `fill!` for both cases? Or is there a reason to prefer `.=` when possible?

Reply:
The original had `.=`, but I think we could just replace it will  only `fill!`. I timed them and I think they do the same thing with numbers.